bgzf.c

```diff
@@ -2205,6 +2205,10 @@ static inline int64_t bgzf_seek_common(BGZF* fp,
                 abort();  // Should not get to any other state
             }
         } while (fp->mt->command != SEEK_DONE);
+
+        if (fp->mt->errcode)
+            return -1;
```

Comment:
I think this should also do `fp->errcode |= BGZF_ERR_IO` and it might be good if `bgzf_mt_seek()` set `mt‑>errcode` to `errno` (instead of `BGZF_ERR_IO`) so that this could also restore `errno` in the caller.

Also maybe this should come after `…‑>command = NONE` and unlocking the mutex, but perhaps continue to not do the resetting of `fp‑>block_*`.

Reply:
Hi @jmarshall, I don't suppose you have a decent bit of code to test a seek error?

Reply:
I'm afraid not. As in, to make a seek error deterministically actually happen so that you can make a test case that exercises it? I've always vaguely meant to write a LD_PRELOAD hack to inject I/O errors, but have never got around to it. But probably such things exist…

Possibly one could make an HTSlib plugin that did this without needing to have a test-only build of _hfile.o_… 🤔